substrate/pkg/controller/substrate/cluster/addons/kubeproxy.go

```diff
@@ -0,0 +1,48 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/awslabs/kit/substrate/pkg/apis/v1alpha1"
+	"github.com/awslabs/kit/substrate/pkg/controller/substrate/cluster"
+	proxyaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type KubeProxy struct {
+}
+
+func (k *KubeProxy) Create(ctx context.Context, substrate *v1alpha1.Substrate) (reconcile.Result, error) {
+	client, err := KubeClientFor(ctx, substrate)
+	if errors.Is(err, ErrWaitingForSubstrateEndpoint) {
```

Comment:
I'm not sure you need this abstraction `ErrWaitingForSubstrateEndpoint` at all if you follow the recommendation of the comment above.

Reply:
The reason I added this abstraction was I didn't want to return 3 results like -
```
func KubeClientFor(ctx context.Context, substrate *v1alpha1.Substrate) (*kubernetes.Clientset, reconcile.Result, error) {
```
I just indicate the caller with `ErrWaitingForSubstrateEndpoint` to requeue or not when there is an error.